src/main/java/com/epam/lowcost/service/implementations/UserServiceImpl.java

```diff
@@ -48,4 +50,33 @@ public User verifyUser(String email, String password) {
         if (user != null && BCrypt.checkpw(password, user.getPassword())) return user;
         return null;
     }
+
+    @Override
+    public Map<String, Object> getUsersByPage(int pageId, int usersByPage) {
+        if (pageId <= 0) {
+            pageId = 1;
+        }
+        int users = userDAO.countUsers();
+        int pagesNum;
+        if (users % usersByPage != 0) {
+            pagesNum = (users / usersByPage + 1);
+        } else {
+            pagesNum = (users / usersByPage);
+        }
+        if (pageId >= pagesNum) {
+            pageId = pagesNum;
+        }
```

Comment:
difficult....

Reply:
Here I check if pageId we got from view is bigger than our max number of pages we will show, if it is we prevent it.